### PR TITLE
Delete Spring related content in Reference.

### PR DESCRIPTION
**Pull request by chenrujun**
Now in https://docs.microsoft.com, we have 2 places to put our content about azure-spring-xxx

1. https://docs.microsoft.com/en-us/java/api/overview/azure/spring-boot-starter-keyvault-secrets-readme?view=azure-java-stable
2. https://docs.microsoft.com/en-us/azure/developer/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-key-vault

After discuss with our PM(@seanli1988 ), we think that keep the 2 both exist in https://docs.microsoft.com will make our customer confused. So we decide to delete the first one.

### CODE REVIEW

File: docs-ref-mapping/reference-latest.yml [hunk 1]
```diff
@@ -703,79 +703,6 @@
...
       children:
       - com.microsoft.azure.management.streamanalytics*
-  - name: Spring Boot
-    uid: azure.java.sdk.landingpage.services.springboot
-    href: ~/docs-ref-services/latest/spring-boot-readme.md
-    landingPageType: Service
-    items:
-    - name: Spring Boot Starter
-      uid: azure.java.sdk.landingpage.services.springboot.starter
-      href: ~/docs-ref-services/latest/spring-boot-starter-readme.md
```
Comment (chenrujun) [inline docs-ref-mapping/reference-latest.yml:713]: I want to delete these md files, but I'm not sure whether there is some job will sync these files from azure-sdk-for-java repo again.

**Comment by chenrujun** [comment]
Hi, @danieljurek , @Taojunshen , @scbedd, @yiliuTo , please help to review this PR.

And do you know how to fix

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,4 @@ _javadocs/
 code2yaml/
 api/.inter/
 .openpublishing.buildcore.ps1
+.idea
```
Comment (yiliuTo) [inline .gitignore:15]: Is is necessary?

**Review by yiliuTo — APPROVED** [review]
LGTM

File: .gitignore (hunk 2 above)
Comment (chenrujun) [reply .gitignore:15]: It's necessary for Intellij users. Let's keep it.

File: docs-ref-mapping/reference-latest.yml (hunk 1 above)
Comment (scbedd) [reply docs-ref-mapping/reference-latest.yml:713]: Hey @chenrujun , We're pulling the entire `spring boot` hierarchy. I'm ok with that, if they exist in another location. What I want to follow-up on is if I can remove the generated content as well.

Comment (chenrujun) [reply docs-ref-mapping/reference-latest.yml:713]: Hi, @scbedd .

Thank you very much for your information.

>  if I can remove the generated content as well.

Yes, please. It's better to remove the generated contents.

Comment (scbedd) [reply docs-ref-mapping/reference-latest.yml:713]: Awesome! I'll take care of this and rip any onboarded spring project out.

**Pull request merged**